testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -2945,3 +2953,59 @@ fn min_gap_between_blocks() {
 
     signer_test.shutdown();
 }
+
+#[test]
+#[ignore]
+/// Test scenario where there are duplicate signers with the same private key
+/// First submitted signature should take precedence
+fn duplicate_signers() {
+    if env::var("BITCOIND_TEST") != Ok("1".into()) {
+        return;
+    }
+
+    tracing_subscriber::registry()
+        .with(fmt::layer())
+        .with(EnvFilter::from_default_env())
+        .init();
+
+    // Disable p2p broadcast of the nakamoto blocks, so that we rely
+    //  on the signer's using StackerDB to get pushed blocks
+    *nakamoto_node::miner::TEST_SKIP_P2P_BROADCAST
+        .lock()
+        .unwrap() = Some(true);
+
+    info!("------------------------- Test Setup -------------------------");
+    let num_signers = 5;
+    let mut signer_stacks_private_keys = (0..num_signers)
+        .map(|_| StacksPrivateKey::new())
+        .collect::<Vec<_>>();
+
+    // First two signers have same private key
+    signer_stacks_private_keys[1] = signer_stacks_private_keys[0];
+    let duplicate_pubkey = Secp256k1PublicKey::from_private(&signer_stacks_private_keys[0]);
+    let duplicate_pubkey_from_copy =
+        Secp256k1PublicKey::from_private(&signer_stacks_private_keys[1]);
+    assert_eq!(
+        duplicate_pubkey, duplicate_pubkey_from_copy,
+        "Recovered pubkeys don't match"
+    );
+
+    let mut signer_test: SignerTest<SpawnedSigner> = SignerTest::new_with_config_modifications(
+        num_signers,
+        vec![],
+        None,
+        |_| {},
+        |_| {},
+        None,
+        Some(signer_stacks_private_keys),
+    );
+
+    signer_test.boot_to_epoch_3();
+    let timeout = Duration::from_secs(30);
+
+    info!("------------------------- Try mining one block -------------------------");
+
+    signer_test.mine_and_verify_confirmed_naka_block(timeout, num_signers);
```

Comment:
I would add a check here with something along these lines.
```rust
        let chunks = test_observer::get_stackerdb_chunks();
        let mut found_signatures = vec![];
        for chunk in chunks
            .into_iter()
            .map(|chunk| chunk.modified_slots)
            .flatten()
        {
            let Ok(SignerMessage::BlockResponse(BlockResponse::Accepted((hash, signature)))) = SignerMessage::consensus_deserialize(&mut chunk.data.as_slice())
            else {
                continue;
            };
            if duplicate_pubkey.verify(&hash, signature).expect("failed to verify signature") {
                found_signatures.push(signature);
            }
        }
        assert_eq!(found_signatures.len(), 2);
        assert_eq!(found_signatures[0], found_signatures[1]);
   ```

Reply:
I'm working on it now. The thing is there are a lot of messages and they don't seem to be in a specific order, so it's going to be more complicated than this

Reply:
Added checks in [bf860f4](https://github.com/stacks-network/stacks-core/pull/5108/commits/bf860f4d0fc41c34abc418632d778cefcc8fc2c3)